### PR TITLE
gh-94393: Remove unneeded module state from _json

### LINKED ISSUE #94393: Remove unneeded module state from `_json` module
The `_json` module got multi-phase initialisation and heap types in GH-17835. However, there is no need for it to have module state. We can remove approx. 50 lines of code, and get rid of the unneeded state allocation.

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-94393

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Also simplify ref.counting by using modern C API for adding the type objects.

File: Modules/_json.c [hunk 1]
```diff
@@ -1815,70 +1802,41 @@ PyDoc_STRVAR(module_doc,
...
-        sizeof(_jsonmodulestate),
-        speedups_methods,
-        _json_slots,
-        _jsonmodule_traverse,
-        _jsonmodule_clear,
-        _jsonmodule_free,
+    .m_base = PyModuleDef_HEAD_INIT,
+    .m_name = "_json",
+    .m_doc = module_doc,
+    .m_size = 0,
```
Comment (encukou) [inline Modules/_json.c]: The initializer can be left out if it's 0, just like the traverse/clear/free functions.

Comment (erlend-aasland) [reply Modules/_json.c]:
```suggestion
```

**Comment by erlend-aasland** [comment]
FYI, I'll merge this in a couple of days.

**Review by corona10 — APPROVED** [review]
lgtm please go ahead.

**Comment by erlend-aasland** [comment]
Thanks for the reviews!

**Pull request merged**